### PR TITLE
Enable FeatureVectorHook for action tasks

### PR DESCRIPTION
**Pull request by jaegukhyun**
### Summary
Same as title.

### How to test
Unit, and integration task will check these changes, since dump_features will always be true for otx eval.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algorithms/action/adapters/mmaction/task.py [hunk 1]
```diff
@@ -423,32 +428,29 @@ def hook(module, inp, outp):
...
-
-        # pylint: disable=unused-argument
-        def dummy_dump_saliency_hook(model, inp, out):
-            saliency_maps.append(None)
-
-        feature_vector_hook = dump_features_hook if dump_features else dummy_dump_features_hook
-        saliency_map_hook = dump_saliency_hook if dump_saliency_map else dummy_dump_saliency_hook
+        if not dump_features:
+            feature_vector_hook: Union[nullcontext, BaseRecordingForwardHook] = nullcontext()
+        else:
```
Comment (sungmanc) [inline src/otx/algorithms/action/adapters/mmaction/task.py]: `dump_features` is always True in the current implementation. Do we need to consider `else`?

File: src/otx/algorithms/action/adapters/mmaction/task.py [hunk 2]
```diff
@@ -423,32 +428,29 @@ def hook(module, inp, outp):
...
                     eval_predictions.extend(result)
                     for _ in range(videos_per_gpu):
                         prog_bar.update()
+                if isinstance(feature_vector_hook, nullcontext):
+                    feature_vectors = [None] * len(mm_dataset)
+                else:
+                    feature_vectors = feature_vector_hook.records
+                if isinstance(saliency_hook, nullcontext):
+                    saliency_maps = [None] * len(mm_dataset)
+                else:
```
Comment (sungmanc) [inline src/otx/algorithms/action/adapters/mmaction/task.py]: `saliency_hook` is always `nullcontext()` here, too. will there be additional implementation?

Comment (jaegukhyun) [reply src/otx/algorithms/action/adapters/mmaction/task.py]: 613418e15e70b1fe53ecaa3b0707134d52555c7d will handle this

File: src/otx/algorithms/action/adapters/mmaction/task.py (hunk 1 above)
Comment (jaegukhyun) [reply src/otx/algorithms/action/adapters/mmaction/task.py]: 613418e15e70b1fe53ecaa3b0707134d52555c7d will handle this

**Review by sungmanc — APPROVED** [review]
LGTM

**Review by eunwoosh — APPROVED** [review]

**Pull request merged**